test/widgets/action_sheet_test.dart

```diff
@@ -861,6 +862,51 @@ void main() {
           ..bodyFields['flag'].equals('read');
       });
     });
+
+    group('CopyTopicLinkButton', () {
+      setUp(() async {
+        TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+          SystemChannels.platform,
+          MockClipboard().handleMethodCall,
+        );
+      });
+
+      Future<void> tapCopyTopicLinkButton(WidgetTester tester) async {
+        await tester.ensureVisible(find.byIcon(ZulipIcons.link, skipOffstage: false));
+        await tester.tap(find.byIcon(ZulipIcons.link));
+        await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
+      }
+
+      group('copies topic link to clipboard', () {
+        testWidgets('FL >= 271 -> link contains "with" operator', (tester) async {
+          final message = eg.streamMessage(stream: someChannel, topic: someTopic);
+          await prepare(channel: someChannel, topic: someTopic,
+            zulipFeatureLevel: eg.recentZulipFeatureLevel);
+          await showFromAppBar(tester, channel: someChannel,
+            topic: TopicName(someTopic), messages: [message]);
+
+          await tapCopyTopicLinkButton(tester);
+          await tester.pump(Duration.zero);
+          final expectedLink = narrowLink(store,
+            TopicNarrow(someChannel.streamId, TopicName(someTopic), with_: message.id)
+          ).toString();
```

Comment:
How about a quick `check(expectedLink.toString().contains('with')).isTrue()`, and `isFalse` in the "FL < 271" test? Just to satisfy a reader that `narrowLink` handles the feature-level condition correctly and they don't need to check that detail in its implementation or tests.

(No need for a TODO to remove that check in the default/current test when clearing out backwards-compatibility code; it won't be as necessary in that future, but it's just one line so the cost is pretty minimal.)